### PR TITLE
Remove the 'New View' sidebar link

### PR DESCRIPTION
**Pull request by janfaracik**
This PR removes the 'New View' sidebar link as it's currently duplicated on the page, once in the tab bar and once in the sidebar. This simplifies the page and makes it easier to tab through. 

### Proposed changelog entries

* Remove the 'New View' sidebar link.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since="TODO")` or `@Deprecated(since="TODO", forRemoval=true)` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@jenkinsci/sig-ux  

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6703"></a>

### CODE REVIEW

File: test/src/test/java/jenkins/model/JenkinsLocationConfigurationTest.java
Comment (janfaracik) [inline test/src/test/java/jenkins/model/JenkinsLocationConfigurationTest.java:130]: I've updated the test as the 'New View' sidebar action is gone now. The updated test instead tests the 'New View' tab bar link. I've had to create a new project at the start of the test to bypass the 'Welcome to Jenkins!' screen.

**Comment by daniel-beck** [comment]
Pinging @uhafner who filed https://issues.jenkins.io/browse/JENKINS-6290

**Comment by uhafner** [comment]
I think the + button has a clear visualization right now (a tool tip wouldn't hurt though). In the old style tabs it was not clear that one can add a new view.

**Review by uhafner — APPROVED** [review]

**Comment by uhafner** [comment]
I wonder if it would also make sense (not in this PR) to remove the edit and delete view actions in the side panel (for the active view). Maybe those can be integrated into the tab header somehow? What do you think?

**Comment by janfaracik** [comment]
> I think the + button has a clear visualization right now (a tool tip wouldn't hurt though). In the old style tabs it was not clear that one can add a new view.

image

Agreed! I've added a tooltip to the + button :) 

> I wonder if it would also make sense (not in this PR) to remove the edit and delete view actions in the side panel (for the active view). Maybe those can be integrated into the tab header somehow? What do you think?

I also agree, ideally I'd like to move actions to the [top app bar](https://weekly.ci.jenkins.io/design-library/AppBar/) and use the sidebar solely for navigation.

Cheers :)

**Comment by daniel-beck** [comment]
Remember that actions removed from the sidepanel are also removed from the breadcrumbs, making them less easily accessible.

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
can we run through ATH?

**Review by timja — APPROVED** [review]
ATH has passed https://github.com/jenkinsci/jenkins/pull/6703

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by daniel-beck** [comment]
Caused [JENKINS-70825](https://issues.jenkins.io/browse/JENKINS-70825).

**Pull request merged**
